yapapi/runner/__init__.py

```diff
@@ -179,6 +196,13 @@ async def map(
         worker: Callable[[WorkContext, AsyncIterator["Task"]], AsyncIterator[Tuple["Task", Work]]],
         data,
     ):
+        """Run computations on providers.
+
+        :param worker: a function that takes a WorkContext object and a list o tasks,
+                       adds commands to the context object and yields committed comments
```

Comment:
Again, this repeats information included in type annotation. But is not really correct: `worker` need not be a function (`Callable` is slightly more general) and `AsyncIterator` is not a `list`. I'd write something like:
```
:param worker: a callable that produces a stream of `Work` objects for a given stream of tasks and a given work context
```

Reply:
The problem is that your description also repeats the type, which is: `Callable[[WorkContext, AsyncIterator["Task"]], AsyncIterator[Tuple["Task", Work]]]`. And `worker` has to return a stream of `Tuple["Task", Work]]`. The developer needs to know that it's necessary to add commands to the context object and then to `yield ctx.commit(task)`. Maybe we should refactor this method?